File: Practica1/Backend-python/app/repositories/user_repository.py
from app.config import get_connection

def find_by_id(id_usuario):
    conn = get_connection()
    cur = conn.cursor()
    cur.execute("SELECT * FROM usuarios WHERE id_usuario = %s", (id_usuario,))
    user = cur.fetchone()
    cur.close()
    conn.close()
    return user

def update_profile(id_usuario, nombre, foto):
    conn = get_connection()
    cur = conn.cursor()
    cur.execute(
        """
        UPDATE usuarios
        SET nombre_completo = %s,
            foto_perfil = %s
        WHERE id_usuario = %s
        """,
        (nombre, foto, id_usuario)
    )
    conn.commit()
    cur.close()
    conn.close()

def find_by_email(email):
    conn = get_connection()
    cur = conn.cursor()
    cur.execute("SELECT * FROM usuarios WHERE correo = %s", (email,))
    user = cur.fetchone()
    cur.close()
    conn.close()
    return user

def create_user(nombre_completo, correo, password_hash, foto_key):
    conn = get_connection()
    cursor = conn.cursor()

    cursor.execute("""
        INSERT INTO usuarios (nombre_completo, correo, password, foto_perfil)
        VALUES (%s, %s, %s, %s)
        RETURNING id_usuario;
    """, (nombre_completo, correo, password_hash, foto_key))

    user_id = cursor.fetchone()[0]

    conn.commit()
    cursor.close()
    conn.close()

    return user_id

File: Practica1/Backend-Nodejs/src/repositories/user.repository.js
const pool = require("../config/db.js");

exports.findById = async (id_usuario) => {
    const result = await pool.query(
        "SELECT * FROM usuarios WHERE id_usuario = $1",
        [id_usuario]
    );
    return result.rows[0];
};

exports.updateProfile = async (id_usuario, nombre, foto) => {
    await pool.query(
        `UPDATE usuarios
         SET nombre_completo = $1,
             foto_perfil = $2
         WHERE id_usuario = $3`,
        [nombre, foto, id_usuario]
    );
};

exports.findByEmail = async (correo) => {
    const result = await pool.query(
        "SELECT * FROM usuarios WHERE correo = $1",
        [correo]
    );
    return result.rows[0];
};

exports.create = async (correo, nombre_completo, password, foto_perfil) => {
    const result = await pool.query(
        `INSERT INTO usuarios (correo, nombre_completo, password, foto_perfil)
         VALUES ($1, $2, $3, $4)
         RETURNING *`,
        [correo, nombre_completo, password, foto_perfil]
    );
    return result.rows[0];
};